tests/infrastructure/downward-metrics.go

```diff
@@ -93,4 +102,141 @@ var _ = DescribeInfra("downwardMetrics", func() {
+		Expect(err).ToNot(HaveOccurred())
+
+		By("checking the VM conditions when the VM is stopped")
+		Eventually(func(g Gomega) {
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			cond := controller.NewVirtualMachineConditionManager().GetCondition(vm, v1.VirtualMachineFailure)
+			g.Expect(cond).ToNot(BeNil())
+			g.Expect(*cond).To(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+				"Type":    Equal(v1.VirtualMachineFailure),
+				"Reason":  Equal(controller.FeatureNotEnabled),
+				"Status":  Equal(k8sv1.ConditionTrue),
+				"Message": ContainSubstring(downwardmetrics.DownwardMetricsNotEnabledError.Error()),
+			}))
+		}, 60*time.Second, 5*time.Second).Should(Succeed())
+
+		By("starting VM")
+		err = virtClient.VirtualMachine(vm.Namespace).Start(context.Background(), vm.Name, &v1.StartOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("checking the VM conditions when the VM is starting")
+		Eventually(func(g Gomega) {
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			cond := controller.NewVirtualMachineConditionManager().GetCondition(vm, v1.VirtualMachineFailure)
+			g.Expect(cond).ToNot(BeNil())
+			g.Expect(*cond).To(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+				"Type":    Equal(v1.VirtualMachineFailure),
+				"Reason":  Equal(common.FailedCreateVirtualMachineReason),
+				"Status":  Equal(k8sv1.ConditionTrue),
+				"Message": ContainSubstring("Failure while starting VMI: %s", downwardmetrics.DownwardMetricsNotEnabledError.Error()),
+			}))
+		}, 60*time.Second, 5*time.Second).Should(Succeed())
+	},
+		Entry("using a disk", libvmi.WithDownwardMetricsVolume("vhostmd")),
+		Entry("using a virtio serial device", libvmi.WithDownwardMetricsChannel()),
+	)
+
+	DescribeTable("should no create a VMI object if the feature is disabled", func(via libvmi.Option) {
+		vm := libvmi.NewVirtualMachine(libvmifact.NewFedora(via))
+
+		By("disabling downwardMetrics feature")
+		tests.DisableDownwardMetrics(virtClient)
+
+		DeferCleanup(func() {
+			By("re-enabling the downwardMetrics feature")
+			tests.EnableDownwardMetrics(virtClient)
+		})
+
+		vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("starting VM")
+		err = virtClient.VirtualMachine(vm.Namespace).Start(context.Background(), vm.Name, &v1.StartOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Consistently(func() bool {
+			_, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			return errors.IsNotFound(err)
+		}, 60*time.Second, 5*time.Second).Should(BeTrue())
```

Comment:
Assert the condition on the VM instead and try once to get the VMI expecting an IsNotFound error as not to prolong the test unnecessarily?